api/specs.proto

```diff
@@ -270,4 +275,19 @@ message ClusterSpec {
 
 	// TaskDefaults specifies the default values to use for task creation.
 	TaskDefaults task_defaults = 7 [(gogoproto.nullable) = false];
+
+	// SecretConfig specifies cluster-level secret configuration settings, such as
+	// maximum number of secrets.
+	SecretsConfig secrets = 8 [(gogoproto.nullable) = false];
+}
+
+// SecretSpec specifies a user-provided secret.
+message SecretSpec {
```

Comment:
It is rebased out of #1511, but since it's not being merged into #1511 (it's going to be merged into master), the changes appear here because #1511 hasn't been merged yet.